src/types/module.rs

```diff
@@ -168,12 +167,12 @@ impl PyModule {
     ///
     /// `__all__` declares the items that will be imported with `from my_module import *`.
     pub fn index(&self) -> PyResult<&PyList> {
-        match self.getattr("__all__") {
+        match self.getattr(__all__(self.py())) {
             Ok(idx) => idx.downcast().map_err(PyErr::from),
             Err(err) => {
                 if err.is_instance_of::<exceptions::PyAttributeError>(self.py()) {
                     let l = PyList::empty(self.py());
-                    self.setattr("__all__", l).map_err(PyErr::from)?;
+                    self.setattr(__all__(self.py()), l).map_err(PyErr::from)?;
```

Comment:
TBH here it looks like we should have a `let __all__ = __all__(self.py())` at the top to avoid looking up the interned value twice.

Reply:
Fixed.